src/app/ApplicationTemplate.Presentation/CoreStartup.cs

```diff
@@ -219,6 +220,57 @@ void ForceUpdate(object sender, EventArgs e)
 		}
 	}
 
+	private void SuscribeToKillSwitch(IServiceProvider serviceProvider)
+	{
+		var killSwitchService = serviceProvider.GetRequiredService<IKillSwitchService>();
+		var navigationController = serviceProvider.GetRequiredService<ISectionsNavigator>();
+
+		killSwitchService.ObserveKillSwitchActivation()
+			.SelectManyDisposePrevious(async (activated, ct) =>
+			{
+				Logger.LogTrace("Kill switch activation changed to {Activated}", activated);
+
+				if (activated)
+				{
+					await OnKillSwitchActivated(ct);
+				}
+				else
+				{
+					await OnKillSwitchDeactivated(ct);
+				}
+			})
+			.Subscribe()
+			.DisposeWith(Disposables);
+
+		async Task OnKillSwitchActivated(CancellationToken ct)
+		{
+			// Clear all navigation stacks and show the kill switch page.
+			// We clear the navigation stack to avoid weird animations once the kill switch is deactivated.
+			foreach (var modal in navigationController.State.Modals)
+			{
+				await navigationController.CloseModal(ct);
+			}
+
+			foreach (var stack in navigationController.State.Sections)
+			{
+				await ClearNavigationStack(ct, stack.Value);
+			}
+
+			await navigationController.NavigateAndClear(ct, () => new KillSwitchPageViewModel());
+
+			Logger.LogInformation("Navigated to kill switch page succesfully.");
+		}
+
+		async Task OnKillSwitchDeactivated(CancellationToken ct)
+		{
+			if (navigationController.GetActiveViewModel() is KillSwitchPageViewModel)
+			{
+				await ExecuteInitialNavigation(ct, serviceProvider);
+				Logger.LogInformation("Kill switch deactivated");
```

Comment:
End all your log message with a period. (This comment applies to multiple places.)